### PR TITLE
Renode sanity check

### PR DESCRIPTION
**Pull request by PiotrZierhoffer**
This pull requests adds Renode support for sanitycheck.

The tests on Renode are enabled for `m2gl025_miv` platform.

To run tests on Renode, you need to have it installed from a package.
Please download Renode from [the GitHub repository](https://github.com/renode/renode) and follow the [build instructions](https://renode.readthedocs.io/en/latest/advanced/building_from_sources.html).

To start tests, run `./scripts/sanitycheck --platform m2gl025_miv -v`

It is still work in progress. Renode hangs on failed tests, we need to solve closing the framework on timeout.

We are very interested in knowing if this approach (via cmake files and console analysis) is right and how should we move forward with enabling more platforms.

### CODE REVIEW

**Comment by SebastianBoe** [comment]
The general approach wrt. the build system looks aligned with qemu, and is therefore OK. There are some code-technical improvements needed, but we can discuss those when this is no longer an RFC.

**Comment by PiotrZierhoffer** [comment]
One question we'll need to address is how to enable Renode for platforms that already have an emulator - like `hifive1`, which can be started on QEMU (it has `EMU_PLATFORM` variable already set).

**Comment by SebastianBoe** [comment]
> One question we'll need to address is how to enable Renode for platforms that already have an emulator - like hifive1, which can be started on QEMU (it has EMU_PLATFORM variable already set).

Hmm, perhaps we could make EMU_PLATFORM into a list. And use some prefixes in the custom target names to allow running either emulator. Or we could have the user choose an emulator through Kconfig/CMake.

**Comment by nashif** [comment]
any instructions on building renode on fedora 29?

**Comment by nashif** [comment]
seems to work fine, you need to fix your commit messages for this to go in.

**Comment by PiotrZierhoffer** [comment]
The previous push from ddf408a to 2752590 was a mistake.

Current state uses the BinaryHandler, improved with the capability of sending a signal directly to the simulator.

112 out of 113 tests pass. The one that does not is also failing on hardware, we are still investigating why that happens. The failing test is `tests/kernel/mem_pool/mem_pool_api/kernel.memory_pool`

**Comment by nashif** [comment]
```
Traceback (most recent call last):
  File "/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/scripts/sanitycheck", line 3082, in <module>
    main()
  File "/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/scripts/sanitycheck", line 3007, in main
    ts.instances)
  File "/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/scripts/sanitycheck", line 2137, in execute
    mg.add_test_instance(i, options.extra_args)
  File "/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/scripts/sanitycheck", line 1106, in add_test_instance
    self.add_goal(ti, type, args)
  File "/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/scripts/sanitycheck", line 1038, in add_goal
    handler = BinaryHandler(instance)
  File "/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/scripts/sanitycheck", line 349, in __init__
    if instance.pid_file != None:
AttributeError: 'TestInstance' object has no attribute 'pid_file'
```

**Comment by nashif** [comment]
> If I understand correctly, to have the first command issue tests for m2gl025_miv I have to add `default: true` to the relevant yml. Is that something we want to have? I didn't find any guidelines for that.

we can make it default once we have everything in place, right now lets leave it as is and call sanitycheck with the platform.

**Comment by PiotrZierhoffer** [comment]
I am currently preparing the Docker image, as we discussed earlier. I will post a link to the relevant pull request when I'm done.

**Comment by PiotrZierhoffer** [comment]
The Docker-related pull request is here: https://github.com/zephyrproject-rtos/docker-image/pull/6

All Mi-V (113) tests pass properly. `./scripts/sanitycheck` started in Docker also works fine.

**Review by nashif — DISMISSED** [review]

File: boards/riscv32/m2gl025_miv/m2gl025_miv.resc [hunk 1]
```diff
@@ -0,0 +1,16 @@
+:name: Mi-V
```
[the remark this answers is not in the document]
Comment (PiotrZierhoffer) [reply boards/riscv32/m2gl025_miv/m2gl025_miv.resc]: done

**Review by nashif — APPROVED** [review]

**Pull request merged**